gmso/formats/lammpsdata.py

```diff
@@ -654,13 +719,437 @@ def _get_ff_information(filename, unit_style, topology):
     for i, pair in enumerate(pair_lines):
         if len(pair.split()) == 3:
-            type_list[i].parameters["sigma"] = (
-                float(pair.split()[2]) * get_units(unit_style)["distance"]
-            )
-            type_list[i].parameters["epsilon"] = (
-                float(pair.split()[1]) * get_units(unit_style)["energy"]
-            )
+            type_list[i].parameters["sigma"] = float(
+                pair.split()[2]
+            ) * get_units(unit_style, "length")
+            type_list[i].parameters["epsilon"] = float(
+                pair.split()[1]
+            ) * get_units(unit_style, "energy")
         elif len(pair.split()) == 4:
             warnings.warn("Currently not reading in mixing rules")
 
     return topology, type_list
+
+
+def _accepted_potentials():
+    """List of accepted potentials that LAMMPS can support."""
+    templates = PotentialTemplateLibrary()
+    lennard_jones_potential = templates["LennardJonesPotential"]
+    harmonic_bond_potential = templates["LAMMPSHarmonicBondPotential"]
+    harmonic_angle_potential = templates["LAMMPSHarmonicAnglePotential"]
+    periodic_torsion_potential = templates["PeriodicTorsionPotential"]
+    periodic_improper_potential = templates["PeriodicImproperPotential"]
+    opls_torsion_potential = templates["OPLSTorsionPotential"]
+    accepted_potentialsList = [
+        lennard_jones_potential,
+        harmonic_bond_potential,
+        harmonic_angle_potential,
+        periodic_torsion_potential,
+        periodic_improper_potential,
+        opls_torsion_potential,
+    ]
+    return accepted_potentialsList
+
+
+def _validate_potential_compatibility(top):
+    """Check compatability of topology object potentials with LAMMPSDATA format."""
+    pot_types = check_compatibility(top, _accepted_potentials())
+    return pot_types
+
+
+def _validate_unit_compatibility(top, base_unyts):
+    """Check compatability of topology object units with LAMMPSDATA format."""
+    # TODO: Check to make sure all units are in the correct format
+    for attribute in ["sites", "bonds", "angles"]:
+        if attribute == "sites":
+            atype = "atom_type"
+        else:
+            atype = attribute[:-1] + "_type"
+        attr_type = getattr(getattr(top, attribute), atype)
+        for parameter in attr_type.parameters:
+            ind_units = re.sub("[^a-zA-Z]+", " ", parameter.dimensions).split()
+            for unit in ind_units:
+                assert unit in base_unyts(unit.dimensions)
+
+
+def _write_header(out_file, top, atom_style):
+    """Write Lammps file header."""
+    out_file.write(
+        "{} written by topology at {} using the GMSO LAMMPS Writer\n\n".format(
+            top.name if top.name is not None else "",
+            str(datetime.datetime.now()),
+        )
+    )
+    out_file.write("{:d} atoms\n".format(top.n_sites))
+    if atom_style in ["full", "molecular"]:
```

Comment:
How will this handle hybrid atom_styles that add bonds and angles to styles such as DPD or ellipsoid (the former from personal use and the latter is in the example directory for lammps).
Also "bond", "angle", and "amoeba" have native bond and angle capability so the lists here need to be updated.

Reply:
> Also "bond", "angle", and "amoeba" have native bond and angle capability so the lists here need to be updated.

The only atom styles supported at this point are the ones we supported in the mbuild lammpswriter. So for now, we just have these from the doc string:
```
atom_style : str, optional, default='full'
        Defines the style of atoms to be saved in a LAMMPS data file.
        The following atom styles are currently supported: 'full', 'atomic', 'charge', 'molecular'
        see http://lammps.sandia.gov/doc/atom_style.html for more information on atom styles.
```

> How will this handle hybrid atom_styles that add bonds and angles to styles such as DPD or ellipsoid (the former from personal use and the latter is in the example directory for lammps).
hybrid is going to be interesting, to say the least. According to the lammps read_data docs, hybrid will look like:
`atom-ID atom-type x y z sub-style1 sub-style2`
So the way to handle that would be for something in the full topology: 
```python
if len(atom_style) > 0 and atom_style[0] == "hybrid":
    style_formatsList = get_styleformats(atom_style[1], top, section_type)
    for style_format in style_formatsList:
        out_file.write(style_format)
```
Something that you iterate through the topology for might look like:
```python
if len(atom_style) > 0 and atom_style[0] == "hybrid":
    style_formatsList = get_styleformats(atom_style[1], top, section_type)
    for site in top.sites:
        for style_format in style_formatsList:
            out_file.write(style_format(site))
```
where atom_style isn't a string, but a list of styles you pass to the writer function.
i.e. `atom_style = ["hybrid", "charge", "full"]`